From patchwork Thu Feb 28 19:48:12 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Louis Taylor <louis@kragniz.eu>
X-Patchwork-Id: 10833637
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 33321180E
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 28 Feb 2019 19:49:51 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 214992FA28
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 28 Feb 2019 19:49:51 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 13C672FA2E; Thu, 28 Feb 2019 19:49:51 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham
	version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A44FE2FA28
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 28 Feb 2019 19:49:50 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731146AbfB1Ttp (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 28 Feb 2019 14:49:45 -0500
Received: from mail-wm1-f68.google.com ([209.85.128.68]:54205 "EHLO
        mail-wm1-f68.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726678AbfB1Ttp (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 28 Feb 2019 14:49:45 -0500
Received: by mail-wm1-f68.google.com with SMTP id e74so10325481wmg.3
        for <kvm@vger.kernel.org>; Thu, 28 Feb 2019 11:49:43 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=kragniz.eu; s=google;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=RJHJp0v8q0EwV9djfl2B0HUEURFjv69zVjwgW24PDFc=;
        b=IpITimNUIv+V0W8Uqqmg/YLXBEsLaGRBsOda2JUBABdbBjOffSAKDKwBexClRIPr3P
         T/rzuqkpd19cRJwtr6JRSbA+IPzItgSGNYx52mX3sCFBp4UFq7YKbgTrQakm1YOmZulH
         YvzA6SdgibJ5Hy9Qt1pcvj5hsmonafR8NO6uw=
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=RJHJp0v8q0EwV9djfl2B0HUEURFjv69zVjwgW24PDFc=;
        b=K2KgGNlJGS5bUSreRSjfY9WDxH3v9ci/iALnbE4dZ6/+O+i5qzdZBLmhzvojySRl4E
         HgLO49XLAkqGCeTrRgE837cnihGnz2YznKaRADD3WROmDONhfEkfwBW03rV3i+S+fTmN
         g0IMGw8sLI6VuhpFzWgLeWYEBOWSSm36bK2zX2OLj0J9jJv203yp8hXQQd85m5+2bRTF
         +r/2GE97myjkt4Fz9LPBRqNLv+ZhJ0jnEKvDDUJAaEPzjrrA3/0SSU4GS+67H0f5IFGb
         Q7G1rhVvrzkqKC9RB4RARnXUXcpT4khaxHzsDxtjOFVxUsjV2tG9ztZs9Ol+d2STLJOw
         69gQ==
X-Gm-Message-State: APjAAAUXtQoKlsBEmoxzQO25tk7OPpfd7DOrkcmDnZBiQI7NOdtQyGTL
        2knov1EDrt+PFx2IapA5jg9HMg==
X-Google-Smtp-Source: 
 AHgI3IbXMI5KkeIg9rjItUHrgzQkuxyNdlblcx3MgAvZqurY9vkMhNDjydtAQuY97AlmYMAarP+vRw==
X-Received: by 2002:a1c:6643:: with SMTP id a64mr857690wmc.90.1551383382697;
        Thu, 28 Feb 2019 11:49:42 -0800 (PST)
Received: from localhost.localdomain ([213.205.198.100])
        by smtp.gmail.com with ESMTPSA id
 y22sm3928323wmj.6.2019.02.28.11.49.41
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Thu, 28 Feb 2019 11:49:42 -0800 (PST)
From: Louis Taylor <louis@kragniz.eu>
To: alex.williamson@redhat.com
Cc: aik@ozlabs.ru, kvm@vger.kernel.org, linux-kernel@vger.kernel.org,
        clang-built-linux@googlegroups.com, ndesaulniers@google.com,
        jflat@chromium.org, Louis Taylor <louis@kragniz.eu>
Subject: [PATCH] vfio/pci: use correct format characters
Date: Thu, 28 Feb 2019 19:48:12 +0000
Message-Id: <20190228194812.30929-1-louis@kragniz.eu>
X-Mailer: git-send-email 2.20.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

When compiling with -Wformat, clang emits the following warnings:

drivers/vfio/pci/vfio_pci.c:1601:5: warning: format specifies type
      'unsigned short' but the argument has type 'unsigned int' [-Wformat]
                                vendor, device, subvendor, subdevice,
                                ^~~~~~

drivers/vfio/pci/vfio_pci.c:1601:13: warning: format specifies type
      'unsigned short' but the argument has type 'unsigned int' [-Wformat]
                                vendor, device, subvendor, subdevice,
                                        ^~~~~~

drivers/vfio/pci/vfio_pci.c:1601:21: warning: format specifies type
      'unsigned short' but the argument has type 'unsigned int' [-Wformat]
                                vendor, device, subvendor, subdevice,
                                                ^~~~~~~~~

drivers/vfio/pci/vfio_pci.c:1601:32: warning: format specifies type
      'unsigned short' but the argument has type 'unsigned int' [-Wformat]
                                vendor, device, subvendor, subdevice,
                                                           ^~~~~~~~~

drivers/vfio/pci/vfio_pci.c:1605:5: warning: format specifies type
      'unsigned short' but the argument has type 'unsigned int' [-Wformat]
                                vendor, device, subvendor, subdevice,
                                ^~~~~~

drivers/vfio/pci/vfio_pci.c:1605:13: warning: format specifies type
      'unsigned short' but the argument has type 'unsigned int' [-Wformat]
                                vendor, device, subvendor, subdevice,
                                        ^~~~~~

drivers/vfio/pci/vfio_pci.c:1605:21: warning: format specifies type
      'unsigned short' but the argument has type 'unsigned int' [-Wformat]
                                vendor, device, subvendor, subdevice,
                                                ^~~~~~~~~

drivers/vfio/pci/vfio_pci.c:1605:32: warning: format specifies type
      'unsigned short' but the argument has type 'unsigned int' [-Wformat]
                                vendor, device, subvendor, subdevice,
                                                           ^~~~~~~~~
The types of these arguments are unconditionally defined, so this patch
updates the format character to the correct ones for unsigned ints.

Link: https://github.com/ClangBuiltLinux/linux/issues/378
Signed-off-by: Louis Taylor <louis@kragniz.eu>
Reviewed-by: Nick Desaulniers <ndesaulniers@google.com>
---
 drivers/vfio/pci/vfio_pci.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/vfio/pci/vfio_pci.c b/drivers/vfio/pci/vfio_pci.c
index a25659b5a5d1..3fa20e95a6bb 100644
--- a/drivers/vfio/pci/vfio_pci.c
+++ b/drivers/vfio/pci/vfio_pci.c
@@ -1661,11 +1661,11 @@ static void __init vfio_pci_fill_ids(void)
 		rc = pci_add_dynid(&vfio_pci_driver, vendor, device,
 				   subvendor, subdevice, class, class_mask, 0);
 		if (rc)
-			pr_warn("failed to add dynamic id [%04hx:%04hx[%04hx:%04hx]] class %#08x/%08x (%d)\n",
+			pr_warn("failed to add dynamic id [%04x:%04x[%04x:%04x]] class %#08x/%08x (%d)\n",
 				vendor, device, subvendor, subdevice,
 				class, class_mask, rc);
 		else
-			pr_info("add [%04hx:%04hx[%04hx:%04hx]] class %#08x/%08x\n",
+			pr_info("add [%04x:%04x[%04x:%04x]] class %#08x/%08x\n",
 				vendor, device, subvendor, subdevice,
 				class, class_mask);
 	}
